subworkflows/local/contaminant_filter.nf

```diff
@@ -107,9 +180,30 @@ workflow CONTAMINANT_FILTER {
     pirna_reads.set { other_cont_reads }
 
     if (params.other_contamination) {
-        BLAT_OTHER ( Channel.value( 'other' ), ch_reference_hairpin, ch_other_contamination)
-        ch_versions = ch_versions.mix(BLAT_OTHER.out.versions)
-        INDEX_OTHER ( BLAT_OTHER.out.filtered_set )
+        // Search which hairpin miRNAs are present in the other data
+        BLAT_OTHER(ch_reference_hairpin, ch_other_contamination)
+
+        // Extract the significant hits
+        ch_program = Channel.of('BEGIN{FS="\t"}{if(\$11 < 1e-5) print \$2;}').collectFile(name:"program.txt")
```

Comment:
Placeholder for removing ch_program if definining it on top of the workflow outside of the if statements